File: sec14_contest_00/python3/problem_I.py
n = int(input())
arr = tuple(int(i) for i in input().split())

even_count,even_index = 0,0
odd_count,odd_index = 0,0


for i,j in enumerate(arr):
	if j%2==0:
		even_index = i
		even_count +=1
	else:
		odd_index = i
		odd_count += 1

print(even_index+1 if even_count==1 else odd_index+1)
'''
same as:
if even_count==1:
	print(even_index+1)
else:
	print(odd_index+1)
'''

File: sec14_contest_00/cpp/problem_I.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{

    int n;
    scanf("%d",&n);

    int odd_count=0,even_count=0;
    int even_index,odd_index;

    int a; // without array
    for(int i=0;i<n;i++){
        scanf("%d",&a);

        if(a%2==0){
            even_count+=1;
            even_index=i;
        }
        else{
            odd_count+=1;
            odd_index=i;
        }
    }

    if(odd_count==1) printf("%d\n",odd_index+1);
    else if(even_count==1) printf("%d\n",even_index+1);

    return 0;
}


// with array

/*
    int n;
    scanf("%d",&n);

    int odd_count=0,even_count=0;
    int even_index,odd_index;

    int a[n]; // array
    for(int i=0;i<n;i++){
        scanf("%d",&a[i]);
    }

    for(int i=0;i<n;i++){
        if(a[i]%2==0){
            even_count+=1;
            even_index=i;
        }
        else{
            odd_count+=1;
            odd_index=i;
        }
    }

    if(odd_count==1) printf("%d\n",odd_index+1);
    else if(even_count==1) printf("%d\n",even_index+1);

*/